lib/index.js

```diff
@@ -863,29 +862,37 @@ class Image extends RustClass {
           onload = img => fetch.emit('ok', img),
           onerror = err => fetch.emit('err', err),
           passthrough = fn => arg => { 
-            (fn||noop)(arg); 
-            this.#fetch = null; 
+            fn(arg) 
+            this.#fetch = null
+          },
+          success = (data, imgSrc=src) => {
+            this.prop("src", imgSrc)
+            if (this.prop("data", data)) onload(this)
+            else onerror(new Error("Could not decode image data"))
           }
-
+          
     if (this.#fetch) this.#fetch.removeAllListeners()
     let fetch = this.#fetch = new EventEmitter()
         .once('ok', passthrough((this.onload || noop).bind(this)))
         .once('err', passthrough((this.onerror || noop).bind(this)))
 
-    let data;
 
-    if (Buffer.isBuffer(src)){
-      [data, src] = [src, '']
-    } else if (typeof src != 'string'){
-      return
-    } else if (/^\s*data:/.test(src)) {
+    if (Buffer.isBuffer(src)) {
+      // already loaded
+      success(src, '')
+    } else if (typeof src != 'string') {
+      onerror(new Error("'src' property value is neither string nor Buffer type.'"))
+    } else if (src.startsWith('data:')) {
       // data URI
       let [header, mime, enc] = src.slice(0, 40).match(/^\s*data:(?<mime>[^;]*);(?:charset=)?(?<enc>[^,]*),/) || []
       if (!mime || !enc){
         throw new Error(`Invalid data URI header`)
-      }else{
-        data = Buffer.from(src.slice(header.length), enc);
-        this.#type = mime
+      } else {
+        let content = src.slice(header.length)
+        if (enc.match('utf-?8')){
```

Comment:
I think I'm going with `if (enc.toLowerCase() != 'base64')` here since there are many text encodings. The old way was wrong in assuming UTF8, but then again how would one load a bitmap as text anyway.